alibi_detect/cd/classifier.py

```diff
@@ -188,41 +182,60 @@ def score(self, X: Union[np.ndarray, list]) -> float:
             splits = self.skf.split(x, y)
 
         # iterate over folds: train a new model for each fold and make out-of-fold (oof) predictions
-        preds_oof, idx_oof = [], []
+        preds_oof_list, idx_oof_list = [], []
         for idx_tr, idx_te in splits:
             x_tr, y_tr, x_te = x[idx_tr], np.eye(2)[y[idx_tr]], x[idx_te]
             clf = tf.keras.models.clone_model(self.model)
             clf.compile(**self.compile_kwargs)
             clf.fit(x=x_tr, y=y_tr, **self.fit_kwargs)
             preds = clf.predict(x_te, batch_size=self.fit_kwargs['batch_size'])
-            preds_oof.append(preds)
-            idx_oof.append(idx_te)
-        preds_oof = np.concatenate(preds_oof, axis=0)[:, 1]
-        idx_oof = np.concatenate(idx_oof, axis=0)
-        drift_metric = self.metric_fn(y[idx_oof], preds_oof)
-        return drift_metric
-
-    def predict(self, X: Union[np.ndarray, list], return_metric: bool = True) \
-            -> Dict[Dict[str, str], Dict[str, Union[int, float]]]:
+            preds_oof_list.append(preds)
+            idx_oof_list.append(idx_te)
+        preds_oof = np.array(np.concatenate(preds_oof_list, axis=0)[:, 1])
+        idx_oof = np.array(np.concatenate(idx_oof_list, axis=0))
+
+        if self.soft_preds:
+            log_losses_ref = preds_oof[y[idx_oof] == 0]
+            log_losses_cur = preds_oof[y[idx_oof] == 1]
+            dist, p_val = ks_2samp(log_losses_ref, log_losses_cur, alternative='greater')
+        else:
+            baseline_accuracy = max(X_ref.shape[0], X.shape[0]) / (X_ref.shape[0] + X.shape[0])  # exp under null
+            n_oof = idx_oof.shape[0]
+            n_correct = (y[idx_oof] == preds_oof.round()).sum()
+            p_val = binom_test(n_correct, n_oof, baseline_accuracy, alternative='greater')
```

Comment:
Is `greater` the correct alternative to use? e.g. accuracy may either increase or decrease in face of drift.

Reply:
Yeah 'greater' is the correct direction. Note here that we're not looking at the accuracy of a label classifier (which indeed might go up or down under drift) but of a domain classifier trained to classify ref/test instances. If it does this worse than the baseline, that it random chance, not because of drift. Only doing better than a baseline (i.e. there was something available to learn) is indicative of drift.

Reply:
ok makes sense, I guess the underlying assumption is that the classifier is always at least as good as a dummy one.

Reply:
I'm not sure we even need to assume that. Only that the classifier is capable of outperforming a dummy one when it is theoretically possible (i.e. when drift has occured).

Reply:
Oh yes that's what I meant.